### ISSUE TITLE
Context output ignores raw_tex when it is alone in a paragraph

### ISSUE THREAD

**Issue by PeterDavidson** [jgm/pandoc#3969]
I have just updated to the latest (source) version of pandoc and have found that since V 1.19.2.1 the treatment of raw_tex has changed.

It seems that a tex command on a line of its own is now ignored in context, but if anything else is on the line then it is included.

If I build the file (test.md)

```
\placefloats

\placefloats

Here \placefloats 

```

with 

```
pandoc.exe -f markdown+raw_tex+latex_macros -t context -o test.tex test.md
```

I get
```
Here \placefloats 
```

When I run it using V 1.19.2.1 I get

```
\placefloats

\placefloats

Here \placefloats
```
as I would expect. The latex writer does include them and the RawInline elements are visible in the json output so it seems that it is a problem with the context writer.

**Comment by jgm** [comment]
You can see why this is happening:

Dev version:
```
% pandoc -t native
\placefloats

\placefloats

Here \placefloats


[RawBlock (Format "latex") "\\placefloats"
,RawBlock (Format "latex") "\\placefloats"
,Para [Str "Here",Space,RawInline (Format "tex") "\\placefloats"]]
```

1.19.2.1:
```
% /usr/local/bin/pandoc -t native
\placefloats

\placefloats

Here \placefloats
[Para [RawInline (Format "tex") "\\placefloats"]
,Para [RawInline (Format "tex") "\\placefloats"]
,Para [Str "Here",Space,RawInline (Format "tex") "\\placefloats"]]
```

The dev version (correctly) parses the `\placefloats` as a raw block, but identifies this as "latex" rather than the generic "tex".  So it doesn't appear in ConTeXt output.  The old version parses it as a raw inline with format "tex", which is included in either LaTeX or ConTeXt.

Possible fixes:

- Always parse unknown block commands as raw "tex" rather than "latex"
- Have the ConTeXt writer print raw "latex", in addition to "tex" and "context"

**Comment by mb21** [comment]
> Always parse unknown block commands as raw "tex" rather than "latex"

sounds reasonable...

**Comment by PeterDavidson** [comment]
I would think that parsing unknown as "tex" would be more consistent. I'm not sure how feasible it would be to maintain a single source document with raw content intended for latex and context, but it would seem wrong to include clearly identified latex in a context writer. 

For now I've got past the problem by writing a simple python filter to change all raw "latex" into raw "context".

In case anyone else needs a stopgap I've included it here

```python
from pandocfilters import toJSONFilter, RawBlock

def convertRaw(key, value, format, meta):
    if key == 'RawBlock' and value[0]=="latex":
        return RawBlock("context",value[1])

if __name__ == "__main__":
    toJSONFilter(convertRaw)
```

**Comment by cacology** [comment]
Wondering about this. I've been using PeterDavidson's filter for awhile now (thank you!) and discovered this issue repeats for org input as well. I.e.

test.org:
```
#+TITLE: Org Test
* Top Level \placefloats
\placefloats

Text \placefloats
```

Calling `pandoc -f org+raw_tex -t context --filter context-filter.py test.org` (`context-filter.py` is just the filter above) gives:
```
\section[title={Top Level },reference={top-level}]

Text
```

whereas, `pandoc -f org+raw_tex -t latex test.org gives:
```
\hypertarget{top-level}{%
\section{\texorpdfstring{Top Level
\placefloats}{Top Level }}\label{top-level}}

\placefloats

Text \placefloats
```

**Comment by jgm** [comment]
I think we should fix both markdown and org readers so that anything that could be either tex or context gets the label "tex", not "latex".  ("begin...end" environments could still be identified as "latex".)

Then one more change would be needed: currently the context writer only matches on "context", not "tex", for block-level content.

@tarlab - would this change in the org reader cause any problems?

**Comment by tarleb** [comment]
I don't expect any problems (other than some filters which will have to
be updated). In fact, if the LaTeX reader is changed, then the org
reader can be left unchanged. The org reader reuses the LaTeX reader
function to produce Inlines from raw tex.

**Comment by tarleb** [comment]
Correction: I was only thinking about inlines. The code for blocks will have to be changed.

One problem that might be that the Org manual only talks about LaTeX, but not about TeX. This change might be confusing, as the following two blocks will(?) start producing different results, while they are currently interchangable:

    # Will become RawBlock (Format "latex") "\\begin{foo}\n\\end{foo}\n"
    #+begin_export latex
    \begin{foo}
    \end{foo}
    #+end_export

    # Will become RawBlock (Format "tex") "\\begin{foo}\n\\end{foo}\n"
    \begin{foo}
    \end{foo}

OTOH, the option to enable/disable raw tex *is* called `tex` in Org.  I'm not sure yet what's the best way here.

**Comment by tarleb** [comment]
I realized I didn't quite answer your question, sorry. Let me try again:

> I think we should fix both markdown and org readers so that anything that could be either tex or context gets the label "tex", not "latex".

There is only LaTeX for Org, raw TeX is unsupported. The parsing of inline commands as TeX is a pandoc-only feature (and should be rewritten some day).  The Org reader should be unaffected by this change. However, we probably should change the Org writer to output raw `tex` as `latex`.

**Comment by jgm** [comment]
Fixed by 13dea94a9128a4caf3fb820bb21cd8176465c82e
